bert_pretrain/src/mosaic_bert.py

```diff
@@ -65,3 +69,124 @@ def create_mosaic_bert_mlm(pretrained_model_name: str = 'bert-base-uncased',
+              "architectures": [
+                "BertForSequenceClassification
+              ],
+              "attention_probs_dropout_prob": 0.1,
+              "classifier_dropout": null,
+              "gradient_checkpointing": false,
+              "hidden_act": "gelu",
+              "hidden_dropout_prob": 0.1,
+              "hidden_size": 768,
+              "id2label": {
+                "0": "LABEL_0",
+                "1": "LABEL_1",
+                "2": "LABEL_2"
+              },
+              "initializer_range": 0.02,
+              "intermediate_size": 3072,
+              "label2id": {
+                "LABEL_0": 0,
+                "LABEL_1": 1,
+                "LABEL_2": 2
+              },
+              "layer_norm_eps": 1e-12,
+              "max_position_embeddings": 512,
+              "model_type": "bert",
+              "num_attention_heads": 12,
+              "num_hidden_layers": 12,
+              "pad_token_id": 0,
+              "position_embedding_type": "absolute",
+              "transformers_version": "4.16.0",
+              "type_vocab_size": 2,
+              "use_cache": true,
+              "vocab_size": 30522
+            }
+   To create a BERT model for classification:
+    .. testcode::
+        from composer.models import create_bert_classification
+        model = create_bert_classification(num_labels=3) # if the task has three classes.
+    Note:
+        This function can be used to construct a BERT model for regression by setting ``num_labels == 1``.
+        This will have two noteworthy effects. First, it will switch the training loss to :class:`~torch.nn.MSELoss`.
+        Second, the returned :class:`.ComposerModel`'s train/validation metrics will be :class:`~torchmetrics.MeanSquaredError` and :class:`~torchmetrics.SpearmanCorrCoef`.
+        For the classifcation case (when ``num_labels > 1``), the training loss is :class:`~torch.nn.CrossEntropyLoss`, and the train/validation
+        metrics are :class:`~torchmetrics.Accuracy` and :class:`~torchmetrics.MatthewsCorrCoef`, as well as :class:`.BinaryF1Score` if ``num_labels == 2``.
+    """
+    if not model_config:
+        model_config = {}
+
+    model_config['num_labels'] = num_labels
+
+    if not pretrained_model_name:
+        pretrained_model_name = 'bert-base-uncased'
+
+
+    config = transformers.AutoConfig.from_pretrained(pretrained_model_name, **model_config)
+    assert transformers.AutoModelForSequenceClassification.from_config is not None, 'AutoModelForSequenceClassification has from_config method'
+
+    config.return_dict = False
+    # Padding for divisibility by 8
+    if config.vocab_size % 8 != 0:
+        config.vocab_size += 8 - (config.vocab_size % 8)
+     
+    if use_pretrained:
+        assert pretrained_checkpoint is not None, "If use_pretrained=True, pretrained_checkpoint cannot be None."
+        model = BertForSequenceClassification(pretrained_checkpoint=pretrained_checkpoint, config=config)
```

Comment:
Should we just get rid of `use_pretrained` (in both create functions) and base this logic on whether `pretrained_checkpoint` was provided?

Reply:
hmmmm, i think that makes sense. Will update.